### PR TITLE
Feature/multiple models

### PR DESCRIPTION
**Pull request by HugoZwaan**

### CODE REVIEW

File: api/helpers.py [hunk 1]
```diff
@@ -10,3 +10,16 @@ def find_last_modified_recording():
 def find_last_modified_result():
     return max(
         glob.iglob(r"./assets/data/results/*.wav"), key=os.path.getmtime)
+
+def get_taco_models():
+    # get real file path
+    # got bs with relative path
```
Comment (AlexAntonides) [inline api/helpers.py:16]: Is dit nodig?

File: api/routes.py [hunk 2]
```diff
@@ -116,3 +116,12 @@ async def audio_to_tacotron_audio_file(file: UploadFile = File(...)):
     except Exception as e:
         print(e)
```
Comment (AlexAntonides) [inline api/routes.py:117]: Print statement?

File: api/routes.py [hunk 3]
```diff
@@ -116,3 +116,12 @@ async def audio_to_tacotron_audio_file(file: UploadFile = File(...)):
...
         raise HTTPException(status_code=500, detail=str(e))
+
+@router.get('/get_models', response_model=List[str])
+def get_possible_models():
+    try:
+        result = controller.get_models()
+        return result
+    except Exception as e:
+        print(e)
```
Comment (AlexAntonides) [inline api/routes.py:126]: Print statement?

File: api/controller.py [hunk 4]
```diff
@@ -20,7 +21,11 @@ def tts_create_audio_from_text(text):
...
 
-def text_to_tacotron_audio_file(text):
+def text_to_tacotron_audio_file(text, model):
     tta = TacoTronAdapter()
-    absolute_file_path = tta.generate_wav(text)
+    absolute_file_path = tta.generate_wav(text, model)
     return absolute_file_path
+
+def get_models():
+    return get_taco_models()
```
Comment (QuirijnLangedijk) [inline api/controller.py:30]: nutteloze functie

File: api/models/TTS_model.py [hunk 5]
```diff
@@ -2,5 +2,5 @@
 
 
 # Model for the taco POST request
-class TextData(BaseModel):
+class TTS_model(BaseModel):
```
Comment (QuirijnLangedijk) [inline api/models/TTS_model.py:5]: _ in classname?

File: api/controller.py (hunk 4 above)
Comment (HugoZwaan) [reply api/controller.py:30]: mvc model

File: api/models/TTS_model.py (hunk 5 above)
Comment (AlexAntonides) [reply api/models/TTS_model.py]: mvc model

File: api/helpers.py [hunk 6]
```diff
@@ -10,3 +10,17 @@ def find_last_modified_recording():
...
+    # to get the full file path
+    dir_path = os.path.dirname(os.path.realpath(__file__)).replace('\\', '/')
+
+    # get folder names with model inside them
+    # step 1: change \\ ==> /
+    # step 2: get everything after checkpoints/ 
+    # step 3: remove everything after underscore like ['_tts', '_raw']
+    model_folders = [i.replace('\\', '/').split('checkpoints/')[1].split('_')[0] for i in glob.iglob(f"{dir_path}/models/checkpoints/*/")]
+
+    # verwijder dubbele
```
Comment (QuirijnLangedijk) [inline api/helpers.py:25]: Engelse comment aub

File: api/helpers.py [hunk 7]
```diff
@@ -10,3 +10,16 @@ def find_last_modified_recording():
 def find_last_modified_result():
     return max(
         glob.iglob(r"./assets/data/results/*.wav"), key=os.path.getmtime)
+
```
[the remark this answers is not in the document]
Comment (AlexAntonides) [reply api/helpers.py:13]: mvc model

File: api/routes.py [hunk 8]
```diff
@@ -107,12 +104,20 @@ async def audio_to_tacotron_audio_file(file: UploadFile = File(...)):
...
         return FileResponse(str(wav_audio_file_path))
     except Exception as e:
-        print(e)
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.get('/get_models', response_model=List[str])
+def get_available_models():
+    try:
+        result = controller.get_models()
```
Comment (QuirijnLangedijk) [inline api/routes.py:120]: return controller.get_models()

Comment (AlexAntonides) [reply api/routes.py:120]: mvc model

**Review by QuirijnLangedijk — APPROVED** [review]

**Review by Xyrai — APPROVED** [review]

**Pull request merged**